docs/concepts/validation_decorator.md

```diff
@@ -39,151 +39,84 @@ except ValidationError as exc:
     """
 ```
 
-## Argument types
+## Parameter types
 
-Argument types are inferred from type annotations on the function, arguments without a type decorator are considered
-as `Any`. All types listed in [types](../concepts/types.md) can be validated, including Pydantic models and
-[custom types](../concepts/types.md#custom-types).
-As with the rest of Pydantic, types can be coerced by the decorator before they're passed to the actual function:
+Parameter types are inferred from type annotations on the function, or as [`Any`][typing.Any] if unannotated . All types listed in [types](types.md) can be validated, including Pydantic models and [custom types](types.md#custom-types).
+As with the rest of Pydantic, types are by default coerced by the decorator before they're passed to the actual function:
 
-```py test="no-print-intercept"
-# TODO replace find_file with something that isn't affected the filesystem
-import os
-from pathlib import Path
-from typing import Optional, Pattern
+```py
+from datetime import date
 
-from pydantic import DirectoryPath, validate_call
+from pydantic import validate_call
 
 
 @validate_call
-def find_file(path: DirectoryPath, regex: Pattern, max=None) -> Optional[Path]:
-    for i, f in enumerate(path.glob('**/*')):
-        if max and i > max:
-            return
-        if f.is_file() and regex.fullmatch(str(f.relative_to(path))):
-            return f
-
+def greater_than(d1: date, d2: date, *, include_equal=False) -> date:  # (1)!
+    if include_equal:
+        return d1 >= d2
+    else:
+        return d1 > d2
 
-# note: this_dir is a string here
-this_dir = os.path.dirname(__file__)
 
-print(find_file(this_dir, '^validation.*'))
-print(find_file(this_dir, '^foobar.*', max=3))
+d1 = '2000-01-01'  # (2)!
+d2 = date(2001, 1, 1)
+greater_than(d1, d2, include_equal=True)
 ```
 
-A few notes:
+1. Because `include_equal` has no type annotation, it will be inferred as [`Any`][typing.Any].
+2. Although `d1` is a string, it will be converted as a [`date`][datetime.date] object.
```

Comment:
```suggestion
2. Although `d1` is a string, it will be converted to a [`date`][datetime.date] object.
```

Reply:
Maybe "by the decorator" as well like was in the previous example?

Reply:
Imo this is a bit redundant, as we already state "... by default coerced by the decorator..." just before the example

Reply:
Sounds good, but I think we should still do the `as` -> `to`.